net: initialize nMessageSize to uint32_t max

Crypt-iQ opened the pull request:
nMessageSize is uint32_t and is set to -1. This will warn with `-fsanitize=implicit-integer-sign-change` when V1TransportDeserializer calls into the ctor.  This pull initializes nMessageSize to `numeric_limits<uint32_t>::max()` instead and removes the ubsan suppression.

practicalswift commented:
Concept ACK

Last time I checked (early April) this was the only remaining `-fsanitize=integer` warning in `protocol.cpp`. In other words you should be able to remove the suppression for `protocol.cpp` as part of this PR :) 

To limit the scope of this PR and to be guarantee preservation of current behaviour I would suggest keeping `std::numeric_limits<uint32_t>::max()` as the initial value. (Not claiming that changing to `0` would change behaviour: I haven't check.)

As laanwj mentioned in-class member initialization would be nice.

Crypt-iQ committed:
net: initialize nMessageSize to max uint32_t instead of -1

nMessageSize is uint32_t and is set to -1. This will warn with
-fsanitize=implicit-integer-sign-change.

promag reviewed (commented):
Code review ACK 9c891b64ffd14bc8216dbd5eb60816043af265b6.

Here or follow up it could even drop the constructor and ditch `memset` calls and just intialize arrays like `char pchCommand[COMMAND_SIZE]{};`

laanwj commented:
Code review ACK 9c891b64ffd14bc8216dbd5eb60816043af265b6

> Here or follow up it could even drop the constructor and ditch memset calls and just intialize arrays like char pchCommand[COMMAND_SIZE]{};

Agree, would be nice to do this, though I think this PR is good (and self-contained) as-is.

Crypt-iQ commented on src/protocol.cpp:
```diff
@@ -91,7 +91,7 @@ CMessageHeader::CMessageHeader()
     memset(pchMessageStart, 0, MESSAGE_START_SIZE);
     memset(pchCommand, 0, sizeof(pchCommand));
-    nMessageSize = -1;
+    nMessageSize = 0;
```
I think I can tackle this in another PR, thanks for the suggestion
